js/components/matrix.js

```diff
@@ -0,0 +1,366 @@
+     *
+     * @param {object} val - value of selected options
+     *
+     * @returns {undefined}
+     */
+    checkValue: function(val) {
+      if (typeof val === 'string' && val !== '') {
+        val = JSON.parse(this.value);
+      }
+
+      var checkFlag = '';
+      var $vm = this;
+      var columnOpt = [];
+
+      if (val === '') {
+        val = {};
+        checkFlag = 'clear';
+      } else if (!_.isEmpty(val)) {
+        var result = [];
+
+        _.forOwn(val, function(value) {
+          if (typeof value !== 'undefined') {
+            result.push(value);
+          }
+        });
+
+        if (result.length > 0) {
+          _.forEach(result, function(col) {
+            _.find($vm.columnOptions, function(column) {
+              if (column.label === col || column.id === col) {
+                columnOpt.push(column);
+              }
+            });
+          });
+        }
+
+        if (result.length === 0) {
+          checkFlag = 'clear';
+        } else if (columnOpt.length === 0) {
+          checkFlag = 'clear';
+        } else {
+          checkFlag = 'set';
+        }
+      } else {
+        checkFlag = 'set';
+      }
+
+      return checkFlag;
+    },
+    onReset: function(data) {
+      if (data.id === this.$parent.id) {
+        if (this.defaultValueSource !== 'default') {
+          this.setValueFromDefaultSettings({ source: this.defaultValueSource, key: this.defaultValueKey });
+          this.$emit('_input', this.name, this.value);
+        }
+
+        if (this.checkValue(this.value) === 'clear') {
+          this.clearValue();
+        }
+      }
+    },
+
+    onBeforeSubmit: function(data) {
+      var $vm = this;
+
+      _.forOwn(data[this.name], function(key, val) {
+        var row = _.find($vm.rowOptions, function(row) {
+          return (_.has(row, 'label') && _.has(row, 'id')) ? row.id === val : row.label === val;
+        });
+
+        if (!row) {
+          delete data[$vm.name][val];
+        }
+      });
+    },
+
+    /**
+     * Check if the value has correct rowOptions and columnOptions or not
+     *
+     * @returns {undefined}
+     */
+    checkInvalidValue: function() {
+      if (!this.value) {
+        return;
+      }
+
+      var validColumns = [];
+      var $vm = this;
+
+      _.forOwn(this.value, function(key) {
+        _.find($vm.columnOptions, function(col) {
+          if (col.label === key || col.id === col) {
+            validColumns.push(col);
+          }
+        });
+      });
+
+      if (validColumns.length > 0) {
+        return true;
+      }
+
+      return false;
+    }
+  },
+  validations: function() {
+    var $vm = this;
+    var rules = {
+      value: {}
+    };
+
+    if (this.required && !this.readonly) {
+      rules.value.required = function() {
+      // Check that every row has a non-empty value
+        return _.every($vm.rowOptions, function(row) {
+          return typeof $vm.value[row.id || row.label] !== 'undefined';
```

Comment:
does empty string entered by user will be also catched here? do you want to trim the cell values for spaces or no?

Reply:
@yogeshmsharma-xoriant here we are allowing user to enter blank line for options like radio and checkbox.